torch/nn/parameter.py

```diff
@@ -138,3 +138,98 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
             'explicitly manipulating `torch.nn.parameter.UninitializedParameter` '
             'objects. When using LazyModules Call `forward` with a dummy batch ' 
             'to initialize the parameters before calling torch functions'.format(func))
+
+
+class UninitializedBuffer(torch.Tensor):
```

Comment:
This looks almost identical the parameter version above. i wonder if we could reduce the code duplication with a `UninitializedMixin` that is shared for both here?